src/typ.rs

```diff
@@ -2205,6 +2216,13 @@ fn infer_let(
     let pattern = PatternTyper::new(env, level).unify(pattern, value_typ)?;
     let then = infer(then, level, env)?;
     let typ = then.typ();
+
+    // Check that any type annotation is accurate.
+    if let Some(ann) = annotation {
+        let ann_typ = env.type_from_ast(ann, &mut hashmap![], NewTypeAction::MakeGeneric)?;
```

Comment:
Is it okay for me to pass in `hashmap![]` as the vector of type variables? I assumed so, based on my understanding of what this function is doing, but I'm a little fuzzy on some of the details